### PR TITLE
Dmitrii_Matvienko Tasks 1-10

### PR DESCRIPTION
**Pull request by MatvienkoDO**

### CODE REVIEW

File: tasks/5_profile-lookup/profile-lookup.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export function lookupProfile(firstName, prop) {
+  const contact = contacts
+      .find((contact) => contact && contact.firstName === firstName);
+
+  if (!contact) {
+    return errors.noSuchContact;
+  }
+
+  if (prop in contact === false) {
```
Comment (vovcyan) [inline tasks/5_profile-lookup/profile-lookup.js]: It is not a good way to use === false or === true in conditions
For example you can replace it by:
if ( !(prop in contact)) or by something else

File: tasks/7_random-fraction/random-fraction.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export function randomFraction() {
+  return Math.random();
```
Comment (vovcyan) [inline tasks/7_random-fraction/random-fraction.js]: Function must return decimal value
For example: 0.1, 0.2, ... , 0.9

File: tasks/9_find-numbers/find-numbers.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export function getNumbers(str) {
+  const regexResult = str.match(/\d/g);
+
+  const numbers = (regexResult || [])
+      .map(Number)
+      .filter((n) => !isNaN(n));
```
Comment (vovcyan) [inline tasks/9_find-numbers/find-numbers.js]: I think it's unnecessary way

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/5_profile-lookup/profile-lookup.js (hunk 1 above)
Comment (MatvienkoDO) [reply tasks/5_profile-lookup/profile-lookup.js]: done

File: tasks/7_random-fraction/random-fraction.js (hunk 2 above)
Comment (MatvienkoDO) [reply tasks/7_random-fraction/random-fraction.js]: done

File: tasks/9_find-numbers/find-numbers.js (hunk 3 above)
Comment (MatvienkoDO) [reply tasks/9_find-numbers/find-numbers.js]: done

File: tasks/8_factorize-number/factorize-number.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  let result = 1;
+
+  for (let i = 2; i <= n; ++i) {
+    result *= i;
+  }
+
+  return result;
+}
+
+/* solution through tail-recursion
```
Comment (alexkond) [inline tasks/8_factorize-number/factorize-number.js:11]: Tail optimization is neat. I like it!

**Review by alexkond — APPROVED** [review]

File: tasks/10_remove-whitespaces/remove-whitespaces.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export function trimWhiteSpaces(text) {
+  return text
+      .split(' ')
```
Comment (vadeneev) [inline tasks/10_remove-whitespaces/remove-whitespaces.js:3]: Have you tried to measure what could be faster - split join or regexp ?

File: tasks/1_golf-score/golf-score.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    HOLE_IN_ONE,
+    EAGLE,
+    BIRDIE,
+    PAR,
+    BOGEY,
+    DOUBLE_BOGEY,
+    GO_HOME,
+  } = GOLF_SCORES;
+
+  if (strokes === 1) {
```
Comment (vadeneev) [inline tasks/1_golf-score/golf-score.js:22]: It works! However, how do you feel about switch case in that particular case ? It looks like you did it, mean switch without switch

File: tasks/2_convert-to-f/convert-to-f.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export function convertToF(temperature) {
+  return temperature * 1.8 + 32;
```
Comment (vadeneev) [inline tasks/2_convert-to-f/convert-to-f.js:2]: For example temperature = 26 results with 78.80000000000001 - how do you like it ? How about rounding values ?

File: tasks/4_stand-in-line/stand-in-line.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export function nextInLine(array, element) {
+  array.push(element);
```
Comment (vadeneev) [inline tasks/4_stand-in-line/stand-in-line.js:2]: What i don't like in this function - is that we are mutating array we passing to it. It is better to use pure functions https://hackernoon.com/javascript-and-functional-programming-pt-3-pure-functions-d572bb52e21c

second thing  - I bet, that .slice(1) would be significantly faster

File: tasks/6_reverse-string/reverse-string.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+export function reverseString(value) {
+  return value
+      .split('')
+      .reverse()
+      .join('');
```
Comment (vadeneev) [inline tasks/6_reverse-string/reverse-string.js:5]: Tests passed.
But, how about `foo 𝌆 bar mañana mañana` ? try it ;)

Comment (vadeneev) [reply tasks/6_reverse-string/reverse-string.js:5]: it is not change request as far as tests passed

File: tasks/5_profile-lookup/profile-lookup.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+];
+
+const errors = {
+  noSuchContact: 'No such contact',
+  noSuchProperty: 'No such property',
+};
+
+export function lookupProfile(firstName, prop) {
+  const contact = contacts
+      .find((contact) => contact && contact.firstName === firstName);
```
Comment (vadeneev) [inline tasks/5_profile-lookup/profile-lookup.js:35]: Oh, right, undefined

**Review by vadeneev — APPROVED** [review]

File: tasks/4_stand-in-line/stand-in-line.js (hunk 8 above)
Comment (vadeneev) [reply tasks/4_stand-in-line/stand-in-line.js:2]: meh, it should mutate

**Review by vadeneev — APPROVED** [review]

**Review by vovcyan — APPROVED** [review]

File: tasks/9_find-numbers/find-numbers.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+export function getNumbers(str) {
+  const regexResult = str.match(/\d/g);
```
Comment (vadeneev) [inline tasks/9_find-numbers/find-numbers.js:2]: Should we treat s123d as a string with one number ?

File: tasks/10_remove-whitespaces/remove-whitespaces.js (hunk 5 above)
Comment (MatvienkoDO) [reply tasks/10_remove-whitespaces/remove-whitespaces.js:3]: I have measured both options. Measurement was showing that split-join is faster. 69 against 107. It means that in average replace(/\s/g, '') takes 1.55 times more time.

Comment (MatvienkoDO) [reply tasks/10_remove-whitespaces/remove-whitespaces.js:3]: https://drive.google.com/file/d/10queSKlPAlLVFUmNRIkibEzHB8INK6kQ/view?usp=sharing

File: tasks/9_find-numbers/find-numbers.js (hunk 11 above)
Comment (MatvienkoDO) [reply tasks/9_find-numbers/find-numbers.js:2]: The task is named 'find numbers'. So, i think that your correction is right

**Pull request merged**